File: app/src/Shared/Application/Bus/CommandHandler.php
<?php

declare(strict_types=1);

namespace Src\Shared\Application\Bus;

/**
 * COMMAND HANDLER (CQRS Pattern)
 * ==============================
 * Command Handler — Command-ı qəbul edib biznes əməliyyatını icra edir.
 *
 * QAYDALAR:
 * 1. Hər Command-ın BİR Handler-i olur (1:1 əlaqə).
 * 2. Handler biznes logikasını koordinasiya edir.
 * 3. Handler birbaşa biznes qaydalarını ehtiva etmir —
 *    o, Domain layer-dəki Entity/Service-ləri çağırır.
 *
 * AXIN:
 * Controller → CreateOrderCommand → CommandBus → CreateOrderHandler
 *   Handler daxilində:
 *   1. DTO-dan domain obyekti yarat
 *   2. Biznes qaydalarını yoxla (Specification)
 *   3. Repository-yə saxla
 *   4. Domain Event-ləri dispatch et
 */
interface CommandHandler
{
    public function handle(Command $command): mixed;
}

File: app-spring/src/main/java/az/ecommerce/shared/application/bus/CommandHandler.java
package az.ecommerce.shared.application.bus;

/**
 * Laravel: src/Shared/Application/Bus/CommandHandler.php
 * Spring: hər Handler @Component (yaxud @Service) işarələnir.
 * CommandBus reflection ilə command tipinə görə uyğun handler-i tapır.
 *
 * NÜMUNƏ:
 *   @Service
 *   public class CreateOrderHandler implements CommandHandler<CreateOrderCommand, UUID> {
 *       public UUID handle(CreateOrderCommand command) { ... }
 *   }
 */
public interface CommandHandler<C extends Command<R>, R> {
    R handle(C command);
}
